projects/rocprofiler-sdk/source/lib/rocprofiler-sdk/counters/metrics.cpp

```diff
@@ -407,6 +413,27 @@ checkValidMetric(const std::string& agent, const Metric& metric)
     return agent_map != nullptr && agent_map->count(base_metric_id) > 0;
 }
 
+bool
+isSupportSpm(const std::string& agent_name, const Metric& metric)
+{
+    auto agents = rocprofiler::agent::get_agents();
+
+    const auto it = std::find_if(agents.begin(), agents.end(), [&agent_name](const auto* agent) {
+        return agent->name == agent_name;
```

Comment:
```suggestion
        return std::string_view{agent->name} == std::string_view{agent_name};
```
Preferable whenever a `const char*` is involved.